llvm/lib/Target/PowerPC/PPCAsmPrinter.cpp

```diff
@@ -493,25 +493,16 @@ static CodeModel::Model getCodeModel(const PPCSubtarget &S,
   return ModuleModel;
 }
 
-static void checkPerGlobalCodeModel(const GlobalValue *GV, MCSymbol *Sym) {
-  // ELF per global code model not supported yet.
-  if (!isa<MCSymbolXCOFF>(Sym))
-    return;
-
+static std::optional<CodeModel::Model>
+hasPerGlobalCodeModel(const GlobalValue *GV) {
```

Comment:
nit , the functionality looks like to get the  CodeModel::Model , change the function name to `getPerGlobalCodeModel`